### PR TITLE
add taf

### PR DESCRIPTION
**Pull request by OlegChub**

### CODE REVIEW

File: src/main/java/models/Content.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Content {
+    public String owner;
+    public boolean share;
+    public int id;
+    public String name;
```
Comment (Fentezi3) [inline src/main/java/models/Content.java:12]: You have the same fields in Dashboard class.
Maybe we can re-use them? Or it's just the same names?

File: src/main/java/models/Page.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Page {
```
Comment (Fentezi3) [inline src/main/java/models/Page.java]: Not clear what exactly page you mean.

File: src/test/java/ApiTests.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import io.qameta.allure.Owner;
+import models.Root;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static helpers.ConfigSetUp.*;
```
Comment (Fentezi3) [inline src/test/java/ApiTests.java]: It's better to import only necessary classes.

**Comment by dKatechev** [comment]
https://github.com/OlegChub/reportPortal/blob/4a79ecc0a95ea3b2bb107732894afb34b331a8af/.github/workflows/maven.yml#L26

Do you need to run tests, release jar and specify default pom file address? Or just check code for compilation?

File: src/main/java/models/Content.java (hunk 1 above)
Comment (OlegChub) [reply src/main/java/models/Content.java:12]: remove unused models

File: src/main/java/models/Page.java (hunk 2 above)
Comment (OlegChub) [reply src/main/java/models/Page.java]: remove unused models

**Comment by OlegChub** [comment]
> https://github.com/OlegChub/reportPortal/blob/4a79ecc0a95ea3b2bb107732894afb34b331a8af/.github/workflows/maven.yml#L26
> 
> Do you need to run tests, release jar and specify default pom file address? Or just check code for compilation?

set mvn test-compile

File: src/test/java/ApiTests.java (hunk 3 above)
Comment (OlegChub) [reply src/test/java/ApiTests.java]: done

File: src/test/resources/config/credentials.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NjUxNDE2NTAsInVzZXJfbmFtZSI6InN1cGVyYWRtaW4iLCJhdXRob3JpdGllcyI6WyJST0xFX0FETUlOSVNUUkFUT1IiXSwianRpIjoiOTQ2ZWU1YzgtMmZkMS00NjkyLTlhOWEtMDU0MWM0OTBmMjA3IiwiY2xpZW50X2lkIjoidWkiLCJzY29wZSI6WyJ1aSJdfQ.xGwaPEx1Gz_sJXq5WrW65HyvWKS17KcWnN6Lx8Yw1bw
```
Comment (dKatechev) [inline src/test/resources/config/credentials.properties]: Are you sure you want to share your Bearer token?

File: src/test/java/ApiTests.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    public void getAllPermittedDashboard() {
+        Root response =
+                given(requestSpecification)
+                        .when()
+                        .get("api/v1/" + getProjectName() + "/dashboard")
+                        .then()
+                        .log().body()
+                        .spec(resSpec)
+                        .extract().as(Root.class);
+        assertEquals(14, response.content.get(0).getId());
```
Comment (dKatechev) [inline src/test/java/ApiTests.java:25]: Magic numbers.
What is 14? 0?

File: src/main/java/models/Root.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Data;
+
+import java.util.ArrayList;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Root {
```
Comment (dKatechev) [inline src/main/java/models/Root.java:10]: Root? Maybe something related to Dashboard, as you use it there?

File: src/main/java/models/Root.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+        public ArrayList<Content> content;
```
Comment (dKatechev) [inline src/main/java/models/Root.java]: Content? Maybe someting more meaningful?

File: src/main/java/config/ProjectConfig.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package config;
+
+import org.aeonbits.owner.Config;
+
+@Config.LoadPolicy(Config.LoadType.MERGE)
+@Config.Sources({
+        "system:properties",
+        "classpath:config/project.properties"
+
```
Comment (dKatechev) [inline src/main/java/config/ProjectConfig.java]: Empty line?

File: src/main/java/config/Credentials.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package config;
+
+import org.aeonbits.owner.Config;
+@Config.LoadPolicy(Config.LoadType.MERGE)
+@Config.Sources({
+        "system:properties",
+        "classpath:config/credentials.properties"
+
```
Comment (dKatechev) [inline src/main/java/config/Credentials.java]: Empty line?

File: src/main/java/config/Credentials.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package config;
+
+import org.aeonbits.owner.Config;
+@Config.LoadPolicy(Config.LoadType.MERGE)
```
Comment (dKatechev) [inline src/main/java/config/Credentials.java:5]: Please separate imports from annotations.

File: src/main/java/config/Credentials.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+@Config.Sources({
+        "system:properties",
+        "classpath:config/credentials.properties"
+
+})
+public interface Credentials extends Config {
+    String token();
+
```
Comment (dKatechev) [inline src/main/java/config/Credentials.java]: Empty line? Please be consistent.

File: src/main/java/helpers/ConfigSetUp.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
+package helpers;
+
+import config.Credentials;
+import config.ProjectConfig;
+import org.aeonbits.owner.ConfigFactory;
+
+public class ConfigSetUp {
+    static ProjectConfig config = ConfigFactory.create(ProjectConfig.class, System.getProperties());
```
Comment (dKatechev) [inline src/main/java/helpers/ConfigSetUp.java]: Maybe add access modifier?

File: src/main/java/specs/RestAssuredSpec.java [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+
+import static helpers.ConfigSetUp.getHost;
+import static helpers.ConfigSetUp.getToken;
+import static io.restassured.RestAssured.with;
+
+public class RestAssuredSpec {
+    static public RequestSpecification requestSpecification =
+            with()
+                    .baseUri(getHost())
+                    .headers("Authorization", "bearer " + getToken())
```
Comment (dKatechev) [inline src/main/java/specs/RestAssuredSpec.java]: Could be replaced with org.apache.http.HttpHeaders.AUTHORIZATION.

File: src/main/java/specs/RestAssuredSpec.java [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+                    .contentType(ContentType.JSON)
+                    .log().all();
+    static public ResponseSpecification resSpec = new ResponseSpecBuilder()
+            .expectStatusCode(200)
```
Comment (dKatechev) [inline src/main/java/specs/RestAssuredSpec.java]: org.apache.http.HttpStatus.SC_OK

File: src/test/java/ApiTests.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+public class ApiTests {
+    @Test
+    @DisplayName("Get all permitted dashboard resources for specified project")
+    @Owner("Oleg Chubryk")
+    public void getAllPermittedDashboard() {
+        Root response =
+                given(requestSpecification)
+                        .when()
+                        .get("api/v1/" + getProjectName() + "/dashboard")
```
Comment (dKatechev) [inline src/test/java/ApiTests.java:20]: logic could be extracted to method with meaningful name

**Comment by dKatechev** [comment]
Please add logging.

File: src/test/resources/config/credentials.properties (hunk 4 above)
Comment (OlegChub) [reply src/test/resources/config/credentials.properties]: Yeah, it's like an example. Also the token becomes invalid after re-login and it expires some time later

File: src/test/java/ApiTests.java (hunk 5 above)
Comment (OlegChub) [reply src/test/java/ApiTests.java:25]: 14 - it is dashboard id, response.content.get(0).getId() refers to json path of this id from the response.
Json file you can find here https://wtools.io/paste-code/bFGa

File: src/main/java/models/Root.java (hunk 6 above)
Comment (OlegChub) [reply src/main/java/models/Root.java:10]: The same thing: it is structure of response  https://wtools.io/paste-code/bFGa

File: src/main/java/models/Root.java (hunk 7 above)
Comment (OlegChub) [reply src/main/java/models/Root.java]: Look at response https://wtools.io/paste-code/bFGa
Classes were created according to the response instances

File: src/main/java/config/ProjectConfig.java (hunk 8 above)
Comment (OlegChub) [reply src/main/java/config/ProjectConfig.java]: removed

File: src/main/java/config/Credentials.java (hunk 11 above)
Comment (OlegChub) [reply src/main/java/config/Credentials.java]: removed

File: src/main/java/helpers/ConfigSetUp.java (hunk 12 above)
Comment (OlegChub) [reply src/main/java/helpers/ConfigSetUp.java]: done

File: src/main/java/specs/RestAssuredSpec.java (hunk 13 above)
Comment (OlegChub) [reply src/main/java/specs/RestAssuredSpec.java]: replaced

File: src/main/java/specs/RestAssuredSpec.java (hunk 14 above)
Comment (OlegChub) [reply src/main/java/specs/RestAssuredSpec.java]: replaced

File: src/test/java/ApiTests.java (hunk 15 above)
Comment (OlegChub) [reply src/test/java/ApiTests.java:20]: don't understand what do u mean

Comment (dsalonka) [reply src/test/java/ApiTests.java:20]: maybe it's better to create service class, e. g DashboardApiService?

File: src/test/java/ApiTests.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static specs.RestAssuredSpec.requestSpecification;
+import static specs.RestAssuredSpec.resSpec;
+
+public class ApiTests {
+    @Test
+    @DisplayName("Get all permitted dashboard resources for specified project")
+    @Owner("Oleg Chubryk")
+    public void getAllPermittedDashboard() {
```
Comment (dsalonka) [inline src/test/java/ApiTests.java:16]: test name about all, in test you call only one

File: src/main/java/config/Credentials.java (hunk 9 above)
Comment (OlegChub) [reply src/main/java/config/Credentials.java]: removed

File: src/main/java/config/Credentials.java (hunk 10 above)
Comment (OlegChub) [reply src/main/java/config/Credentials.java:5]: done

File: src/test/java/ApiTests.java (hunk 16 above)
Comment (OlegChub) [reply src/test/java/ApiTests.java:16]: it is the description of the method from Swagger
image

Comment (dsalonka) [reply src/test/java/ApiTests.java:16]: yes but you test not all of them, so I suggest more specific name according to logic

Comment (OlegChub) [reply src/test/java/ApiTests.java:16]: This test is written nominally, just to be. My goal was not to create real test.

Comment (dsalonka) [reply src/test/java/ApiTests.java:16]: okay then, I see

File: src/test/resources/config/credentials.properties (hunk 4 above)
Comment (dKatechev) [reply src/test/resources/config/credentials.properties]: Still, please, provide a placeholder instead.

File: src/test/resources/config/project.properties [hunk 17]
```diff
@@ -0,0 +1,2 @@
+host=http://localhost:8080/
+projectName=superadmin_personal
```
Comment (dKatechev) [inline src/test/resources/config/project.properties:2]: Are you sure, it is necessery to put this into properties?

File: src/test/java/ApiTests.java (hunk 5 above)
Comment (dKatechev) [reply src/test/java/ApiTests.java:25]: So, you can:
- create constant with meaningful name;
- filter content by some values, more related to the test.

File: src/main/java/models/Root.java (hunk 6 above)
Comment (dKatechev) [reply src/main/java/models/Root.java:10]: So why Root?

**Pull request merged**